pydy/joints.py

```diff
@@ -0,0 +1,136 @@
+__all__ = ['PinJoint', 'SlidingJoint', 'CylindricalJoint', 'SphericalJoint', 'PlanarJoint']
+
+
+class Joint(object):
+    """Serves as a base class for all specific Joints. Contains all public methods
+    to be used by other specific Joint's implementation.
+
+    Note: Some of these methods can be shifted to body. Want to have a discussion
+    on this.
+    """
+    def __init__(self, name=None, parent, par_point_vec_tuple, child, child_point_vec_tuple):
+        if name is None:
+            self._create_name()
+        else:
+            self.name = name
+        self.parent = parent
+        self.child = child
+        self.parent_joint_vector = par_point_vec_tuple
+        self.child_joint_vector = child_point_vec_tuple
+        self.counter = 0
```

Comment:
Unless you make this a class level attribute, you will not ensure unique names in a given Python session, no?